site/app/libraries/database/DatabaseQueries.php

```diff
@@ -5171,6 +5186,34 @@ public function getInstructorLevelAccessCourse(string $user_id): array {
         return $this->submitty_db->rows();
     }
 
+    /**
+     * @return array<Course>
+     */
+    public function getSelfRegisterCourses(string $user_id): array {
+        $query = <<<SQL
+SELECT c.*, t.name AS term_name FROM courses c, terms t WHERE NOT c.self_registration_type = 0 AND c.course NOT IN (
```

Comment:
I'm not sure, this works as is, but if it's not 'best practice' I'm not the most versed in SQL semantics so advice is welcome.

Reply:
I suspect that it works for you locally because there is only one term.  If I'm understanding things correctly here, I believe this will return every combination of courses and terms, which just so happens to work correctly if you only have one term in the database.  Instead, you probably want something like this: 

```
SELECT c.*, t.name as term_name
FROM courses_users u
INNER JOIN courses c ON u.course=c.course AND u.term=c.term
INNER JOIN terms t ON u.term=t.term_id
WHERE u.user_id = ?
```

You can search for various parts of this query in the codebase to see other places where similar queries are executed for reference.

Reply:
I think i have fixed it to only select the term that the course is a part of.